solidity/ecdsa/deploy/00_resolve_random_beacon.ts

```diff
@@ -0,0 +1,34 @@
+import type {
+  HardhatRuntimeEnvironment,
+  HardhatNetworkConfig,
+} from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { getNamedAccounts, deployments, helpers } = hre
+  const { log } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const RandomBeacon = await deployments.getOrNull("RandomBeacon")
+
+  if (RandomBeacon && helpers.address.isValid(RandomBeacon.address)) {
+    log(`using external RandomBeacon at ${RandomBeacon.address}`)
+  } else if (
+    !hre.network.tags.allowStubs ||
+    (hre.network.config as HardhatNetworkConfig)?.forking?.enabled
```

Comment:
Not blocking this PR because we use this pattern everywhere but I think we should at least explain in a comment this condition - it's pretty complex for someone who does not understand mainnet fork tests.